src/overlays/actors/ovl_En_Kgy/z_en_kgy.c

```diff
@@ -41,105 +63,1132 @@ const ActorInit En_Kgy_InitVars = {
+            this->unk_29C &= ~4;
+        }
+    }
+}
+
+void func_80B411DC(EnKgy* this, GlobalContext* globalCtx, s32 arg2) {
+    Player* player = PLAYER;
+
+    switch (arg2) {
+        case 0:
+            this->unk_2B4 = this->unk_2A8;
+            this->actor.focus.pos = this->unk_2A8;
+            this->unk_2E0 = 0;
+            break;
+
+        case 1:
+            if (this->unk_2A0 != NULL) {
+                this->unk_2B4 = this->unk_2A0->actor.focus.pos;
+                this->actor.focus.pos = this->unk_2A0->actor.focus.pos;
+            }
+            this->unk_2E0 = 1;
+            break;
+
+        case 2:
+            if (this->unk_2A4 != NULL) {
+                this->unk_2B4 = this->unk_2A4->actor.world.pos;
+                this->actor.focus.pos = this->unk_2A4->actor.focus.pos;
+            }
+            this->unk_2E0 = 2;
+            break;
+
+        case 3:
+            this->unk_2B4 = player->actor.world.pos;
+            if (this->unk_2A0 != NULL) {
+                this->actor.focus.pos = this->unk_2A0->actor.focus.pos;
+            }
+            this->unk_2E0 = 3;
+            break;
+
+        case 4:
+            if (this->unk_2A0 != NULL) {
+                this->unk_2B4 = this->unk_2A0->actor.focus.pos;
+                this->actor.focus.pos = this->unk_2A0->actor.focus.pos;
+            }
+            this->unk_2E0 = 4;
+            break;
+    }
+}
+
+void func_80B41368(EnKgy* this, GlobalContext* globalCtx, s32 arg2) {
+    ActorCutscene_Stop(this->unk_2D4[this->unk_2E0]);
+    func_80B411DC(this, globalCtx, arg2);
+    this->unk_2E6 = 20;
+    this->unk_29C |= 2;
+}
+
+void func_80B413C8(EnKgy* this) {
+    Vec3f sp1C = this->unk_2A8;
+
+    this->unk_2B4 = sp1C;
+    this->actor.focus.pos = sp1C;
+    ActorCutscene_Stop(this->unk_2D4[this->unk_2E0]);
+    this->unk_2E0 = -1;
+    this->unk_2E6 = 0;
+    this->unk_29C &= ~2;
+}
+
+s32 func_80B41460(void) {
+    if ((gSaveContext.playerForm != PLAYER_FORM_HUMAN) ||
+        ((CUR_BTN_ITEM(BTN_ITEM_B) != ITEM_SWORD_KOKIRI) && (CUR_BTN_ITEM(BTN_ITEM_B) != ITEM_SWORD_RAZOR) &&
+         (CUR_BTN_ITEM(BTN_ITEM_B) != ITEM_SWORD_GILDED))) {
+        return 0xC38;
+    }
+
+    if ((gSaveContext.day % 5) == 3) {
+        return 0xC39;
+    }
+
+    return 0xC3A;
+}
+
+s32 func_80B41528(GlobalContext* globalCtx) {
+    if (CUR_BTN_ITEM(BTN_ITEM_B) == ITEM_SWORD_GILDED) {
+        return 0xC4C;
+    }
+
+    if (CUR_BTN_ITEM(BTN_ITEM_B) == ITEM_SWORD_RAZOR) {
+        return 0xC45;
+    }
+
+    return 0xC3B;
+}
+
+void func_80B415A8(GlobalContext* globalCtx, Vec3f* arg1) {
+    static EffShieldParticleInit D_80B43298 = {
+        16,
+        { 0, 0, 0 },
+        { 0, 200, 255, 255 },
+        { 255, 255, 255, 255 },
+        { 255, 255, 128, 255 },
+        { 255, 255, 0, 255 },
+        { 255, 64, 0, 200 },
+        { 255, 0, 0, 255 },
+        2.0f,
+        15.0f,
+        10.0f,
+        4,
+        { 0, 0, 0, { 0, 128, 255 }, 0, 300 },
+        0,
+    };
```

Comment:
Does this have to be in the function? It's a bit big IMO.

Reply:
It does, very different code gen if it's moved outside.